app/data_models/progress_store.py

```diff
@@ -62,10 +70,10 @@ def _build_map(section_progress_list: Iterable[ProgressDictType]) -> MutableMapp
 
         return {
             (
-                section_progress["section_id"],
-                section_progress.get("list_item_id"),
-            ): Progress.from_dict(section_progress)
-            for section_progress in section_progress_list
+                progress["section_id"],
```

Comment:
Is this not less readable now? I mean, we still track section progress (and this will be less obvious). I still need to check this code some more but are list items (their progress) part of section and its progress? This name sort of suggests it's an instance of Progress class? Do other people have an opinion?

Reply:
We'd had a discussion about naming [here](https://github.com/ONSdigital/eq-questionnaire-runner/pull/1106#discussion_r1234155990). I'm happy with it being very verbose e.g. `section_or_list_item_progress`, or brief and generic `progress`, with the docs throughout the ProgressStore explaining it is not for only sections, it is for sections, repeating sections and list items.

Reply:
Regarding "are list items (their progress) part of section and its progress?":

List item progress is to track if a list item with repeating blocks has been fully completed or partially completed. 
A list item with list item id "item1" from a list collector within section "section1", will be keyed in the ProgressStore as `("section1", "item1")`. The `Progress` object for the `_progress` entry keyed by `("section1", "item1")`, will have the `block_ids` attribute populated by the completed repeating block ids for this list item. Its `status` attribute will be `CompletionStatus.IN_PROGRESS` if the list item has been added via the list collector's add block, but not all repeating blocks have been completed and added to the `block_ids` attribute of the relevant Progress object. The `status` attribute will be `CompletionStatus.COMPLETED` when all repeating blocks have been completed and their ids added to the `block_ids` attribute of the relevant Progress object.

Section and Repeating Section progress works as it always has - there is no conflict in using progress store to track the list item progress. 

All of the renaming of methods and variables has been made to make it more inclusive of the elements for which it tracks progress.

Reply:
Regarding `section_or_list_item_progress` the problem here is list item is an item from a list in a list store, it's not meant to have progress. Maybe it will be better if we use `repeating blocks`?

Reply:
I like the clarity of it being exactly whats stored in it.. `section_or_repeating_blocks_progress`?

Reply:
I think we need to introduce a concept of List Items being complete or incomplete now, because routing out of List Collectors and rendering now depend on know if the List Items are complete (have their repeating blocks been completed) So the repeating blocks are the metric by which a List Item is deemed complete, just like how sections are already, they track completeness by which blocks within them have been completed.

Reply:
<img width="392" alt="Screenshot 2023-06-21 at 13 06 14" src="https://github.com/ONSdigital/eq-questionnaire-runner/assets/114991656/17e65f83-955d-48ba-b09d-c0cb0bb745b8">
I get the dilemma, if you for some reason had repeating sections that used the same list, that list item id appears in a key for both the repeating blocks, and that latter section, so I wouldn't say the list item is done as such, but the repeating blocks are

Reply:
so it could have the additional context of `section_or_list_item_repeating_blocks_progress`

Reply:
we could always roll with it as is and change it later

Reply:
renamed to section_or_repeating_blocks as result of vote 👍